### ISSUE TITLE
Validation messages does NOT respond to 'rules' changes.

### ISSUE THREAD

**Issue by HashemQolami** [ElemeFE/element#1317]
### ElementUI version
1.0.2

### OS/Browers version
Ubuntu / Chrome 51

### Vue version
2.0.8

### Reproduction Link
http://jsfiddle.net/nprj0tb2/

### Steps to reproduce
1. Click the "Submit" button (leave the form empty) in order for the error message be showed up.
2. Click the "Change Locale to ..." button, so that it updates the `rules` object as a `computed` property.
3. Notice the error message which is not updating automatically based on the new rule object.
4. Repeat step 1 again.
5. The error message will be updated manually.

### What is Expected?
if the user changes the language of a multilingual app while some validation errors are on the page, it is surprising that the validation errors are not updated automatically based on the selected language.

In short, the expected behavior is that the error messages update once the `rules` object changes (the validation process should re-run in that case).

### What is actually happening?
Error messages does NOT respond to `rules` object changes.

**Comment by HashemQolami** [comment]
I can send a PR to add this feature if you will.

**Comment by Leopoldthecoder** [comment]
Sure, a PR is definitely welcome.
